ChangeLog

```diff
@@ -19,6 +19,12 @@ Release date: TBA
 
   Refs pylint-dev/pylint#9442
 
+* Make ``astroid.interpreter._import.util.is_namespace`` only consider modules
+  with ``submodule_search_locations`` a namespace. This fixes a problems that ``six.moves``
+  brain was not effective if ``six.moves`` was already imported.
```

Comment:
Some rewording here would be good given that we didn't touch `submodule_search_locations` after all.

Reply:
Oh right, I forgot about updating changelog. I just pushed changes to fix this.